extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,261 @@
+							checked={ showImage }
+							onChange={ () => setAttributes( { showImage: ! showImage } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Title', 'jetpack' ) }
+							checked={ showTitle }
+							onChange={ () => setAttributes( { showTitle: ! showTitle } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Author/Seller', 'jetpack' ) }
+							checked={ showSeller }
+							onChange={ () => setAttributes( { showSeller: ! showSeller } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Price', 'jetpack' ) }
+							checked={ showPrice }
+							onChange={ () => setAttributes( { showPrice: ! showPrice } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Purchase Button', 'jetpack' ) }
+							checked={ showPurchaseButton }
+							onChange={ () => setAttributes( { showPurchaseButton: ! showPurchaseButton } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Color Settings', 'jetpack' ) }
+						colorSettings={ [
+							{
+								value: backgroundColor,
+								onChange: newBackgroundColor =>
+									setAttributes( { backgroundColor: newBackgroundColor } ),
+								label: __( 'Background Color', 'jetpack' ),
+							},
+							{
+								value: textColor,
+								onChange: newTextColor => setAttributes( { textColor: newTextColor } ),
+								label: __( 'Text Color', 'jetpack' ),
+							},
+							{
+								value: buttonAndLinkColor,
+								onChange: newButtonAndLinkColor =>
+									setAttributes( { buttonAndLinkColor: newButtonAndLinkColor } ),
+								label: __( 'Button & Link Color', 'jetpack' ),
+							},
+						] }
+					>
+						{
+							<ContrastChecker
+								{ ...{
+									isLargeText: false,
+									textColor: textColor,
+									backgroundColor: backgroundColor,
+								} }
+							/>
+						}
+					</PanelColorSettings>
+				</>
+			) }
+		</InspectorControls>
+	);
+
+	const blockPreview = () => {
+		const {
+			title,
+			authors,
+			artists,
+			actors,
+			manufacturer,
+			detailPageUrl,
+			listPrice,
+			imageUrlMedium,
+			imageWidthMedium,
+			imageHeightMedium,
+			authorshipInfo,
+		} = data.products.filter( productDataItem => productDataItem.asin === asin )[ 0 ];
+
+		// TODO - we should be able to get this from API in a neater way once we have access
+		const seller = authors.length
+			? authors
+			: artists.length
+			? artists
+			: actors.length
+			? actors
+			: manufacturer
+			? manufacturer
+			: authorshipInfo
+			? authorshipInfo
+			: null;
+
+		// TODO - we have different image sizes in the API
+		const image = imageUrlMedium && (
+			<a href={ detailPageUrl }>
+				<img
+					alt={ title }
+					src={ imageUrlMedium }
+					width={ imageWidthMedium }
+					heigth={ imageHeightMedium }
+				/>
+			</a>
+		);
+
+		const buttonTextColor = tinycolor
+			.mostReadable( buttonAndLinkColor, [ '#ffffff' ], {
+				includeFallbackColors: true,
+				size: 'small',
+			} )
+			.toHexString();
+
+		return (
+			asin && (
```

Comment:
Could we remove a level of nesting here by returning `null` when there's no `asin` set? It also shouldn't have got here without one, but it's good to check.